### ISSUE TITLE
Add options to use pprint.

### ISSUE THREAD

**Issue by eliazar-sll** [gaogaotiantian/watchpoints#27]
I suggest to add an option to use pprint because pprint is good at printing data structures. It is also customizable for different situations. 

For example, in nested list and dictionary they can customize the length with `compact` argument or the indent with the `indent` argument. Another Example, in dictionaries they can print the dictionary with an option to sort it with `sort_dicts`.

I hope you can consider this suggestion.

**Comment by gaogaotiantian** [comment]
I do believe that ``objprint`` can do almost everything ``pprint`` can. If it can't, I think we should add features to ``objprint``(which is a project I own).

The major issue with ``pprint`` and it's cousin ``print`` is that they can only produce garbage when it comes to an object that's not built-in(technically, some built-in objects do not have a good representation either).

What we should probably add is an easier interface for the users to customize their way to print objects so anyone can do their own thing. There is a callback option now but it's rather complicated(even though more thorough). I'll see if I can do something that keeps the current structure while providing a possibility for the users to do their own print.

**Comment by eliazar-sll** [comment]
> What we should probably add is an easier interface for the users to customize their way to print objects so anyone can do their own thing.
I agree! I just realized that there's a `callback` option and it looks a little bit complicated. 
Thank you for your response. Please let me know if can I help regarding this.

**Comment by gaogaotiantian** [comment]
It should be a rather easy implementation, I just need to figure out the structure. Thanks for offering help! I hope we can collaborate on the project in the future!

**Comment by eliazar-sll** [comment]
Great! I look forward with that! I really appreciate your effort.

**Comment by gaogaotiantian** [comment]
Okay I merged #28 which should allow you to use any printer you like. I may have to clean up some other code, but at this point, ``pprint`` should work.
